apps/condo/domains/notification/tasks.js

```diff
@@ -100,6 +101,53 @@ async function deliverMessage (messageId) {
         sender: message.sender,
     }
 
+    if (message.organization) {
+        const messageOrganizationBlackListWhere = {
+            type: message.type,
+            OR: [
+                { organization_is_null: true },
+                { organization: { id: message.organization.id } },
+            ],
+        }
+        const messageOrganizationBlackListRules = await MessageOrganizationBlackList.getAll(keystone, messageOrganizationBlackListWhere)
+
+        if (!isEmpty(messageOrganizationBlackListRules)) {
+            return await Message.update(keystone, message.id, {
+                ...baseAttrs,
+                status: MESSAGE_ERROR_STATUS,
+                processingMeta: {
+                    dv: 1,
+                    error: MESSAGE_TYPE_IN_ORGANIZATION_BLACK_LIST,
+                },
+            })
+        }
+    }
+
+    const messageUserBlackListWhere = {
+        OR: [
+            message.user && { user: { id: message.user.id } },
+            message.phone && { phone: message.phone },
+            message.email && { email: message.email },
+        ].filter(Boolean),
+        deletedAt: null,
+    }
+    const messageUserBlackListRules = await MessageUserBlackList.getAll(keystone, messageUserBlackListWhere)
+
+    if (!isEmpty(messageUserBlackListRules)) {
+        const isMessageTypeInUserBlackList = messageUserBlackListRules.find(rule => rule.type === message.type || isEmpty(rule.type))
```

Comment:
looks like this logic can be moved to helper

Reply:
moved to `checkMessageTypeInBlackList` helper